### DISCUSSION TITLE
Addresable led strip WS2812 guide.

### DISCUSSION THREAD

**Discussion by KondratokGK** [EdgeTX/edgetx#5547] [Q&A]
I want to add WS2812 to my T20. How to add PWM pin to device configuration or something like that.

**Comment by pfeerick** [comment]
You would need to add a code block similar to this, but for the `RADIO_T20`, and would need to configure it appropriately for that target. i.e. you need a suitable free GPIO, Timer, and DMA handler. After that, the rest of the code should pretty much automatically work, as it is looking for `LED_STRIP_GPIO` to be defined to enable the rest of the necessary support. 

https://github.com/EdgeTX/edgetx/blob/e303b126806ce4a112d43fb3116aca29762f0a8f/radio/src/targets/taranis/hal.h#L2046-L2059

**Comment by 3djc** [comment]
Beware that it was not added because the internal 5v rail cannot power it, so you will need to add a power circuit
